### PR TITLE
Always allow JSON fragments

### PR DESCRIPTION
**Pull request by chrisballinger**
This updates the public versions of `JSONEncoder` and `JSONDecoder` to match the behavior shipped with the stdlib in iOS 13. More specifically, always passing `.fragmentsAllowed` to `JSONSerialization`'s reading/writing options.

For more information see:
* https://bugs.swift.org/browse/SR-6163
* https://drewag.me/posts/2019/09/11/json-encoder-change-in-swift-5

Resolves SR-6163.

### CODE REVIEW

File: stdlib/public/Darwin/Foundation/JSONEncoder.swift [hunk 1]
```diff
@@ -1205,7 +1194,7 @@ open class JSONDecoder {
     open func decode<T : Decodable>(_ type: T.Type, from data: Data) throws -> T {
         let topLevel: Any
         do {
-           topLevel = try JSONSerialization.jsonObject(with: data)
+           topLevel = try JSONSerialization.jsonObject(with: data, options: [.fragmentsAllowed])
```
Comment (ollieatkinson) [inline stdlib/public/Darwin/Foundation/JSONEncoder.swift]: Is there a reason to why `options` is not configurable? I'd imagine there are some use-cases where you do not want to allow decoding fragments.

Comment (ollieatkinson) [inline stdlib/public/Darwin/Foundation/JSONEncoder.swift]: Is there a reason to why `options` is not configurable? I'd imagine there are some use-cases where you do not want to allow decoding fragments.

**Comment by parkera** [comment]
@shahmishal Are we missing something here in getting the changes properly merged to GitHub?

**Comment by shahmishal** [comment]
Let me start testing, after that we can merge it if you're ok with the changes.

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by parkera** [comment]
What I mean is that there were probably other changes made that should have been merged here, but it doesn't appear it ever happened.

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by benrimmington** [comment]
This pull request can be closed. The issue was fixed in apple/swift#30615 for Darwin.

SR-6163 can remain open for [swift-corelibs-foundation](https://github.com/apple/swift-corelibs-foundation), which doesn't support the `fragmentsAllowed` options yet.

**Pull request closed without merging**